### ISSUE TITLE
`eslint --env es6` reports `fn(yield);` in generator function as "Unexpected token )" error

### ISSUE THREAD

**Issue by msmhrt** [eslint/eslint#2389]
Hi all,

Please see the following log.

```
$ cat test-yield.js
(function() {
    "use strict";

    let fn = () => {};

    function* testYield() {
        fn(yield);
    }

    testYield().next();
}());
$ eslint --version
v0.19.0
$ eslint --env es6 test-yield.js

test-yield.js
  7:17  error  Unexpected token )

✖ 1 problem (1 error, 0 warnings)

$
```

`eslint --env es6` reports `fn(yield);` in generator function as "Unexpected token )" error, but I think it that the code is valid and `eslint --env es6` should not report errors.

**Comment by michaelficarra** [comment]
This looks like an espree bug. Can you open this issue there instead?

**Comment by hzoo** [comment]
Is it because `fn(yield)` isn't currently supported but `fn(yield a)` is?

**Comment by msmhrt** [comment]
eslint reports error because espree doesn't support yet  `fn(yield)` ?
In case of `fn(yieldd a)`, it seems that eslint reports no error.

**Comment by nzakas** [comment]
Filed: https://github.com/eslint/espree/issues/134

**Comment by nzakas** [comment]
Espree has been updated and a new release made with this fix. If you reinstall ESLint, you should be all set.

**Comment by msmhrt** [comment]
Hi @nzakas,
I saw your patch and I guess `fn(yield, 1)`, `fn([yield])` etc. will become error.
I think it that `&& !match(")")` is only a temporary solution and the problem of YieldExpression consisting of only a yield keyword has been not fixed yet.

**Comment by nzakas** [comment]
@msmhrt it would be helpful if you would for such information with Espree instead of here. Can you open a new open Espree issue and list all the fading cases

**Comment by nzakas** [comment]
Oops, hit the button to soon. The original bug reported here has been fixed. Please open an Espree bug with other errors you find.
